src/main/java/mx/erickb/shipping/service/RoutesManagerService.java

```diff
@@ -0,0 +1,47 @@
+package mx.erickb.shipping.service;
+
+import mx.erickb.shipping.util.RoutesGraph;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class RoutesManagerService implements IRoutesManagerService {
+
+    // Breadth-first search using priority queue
+    public Optional<List<String>> findOptimalRoute(RoutesGraph routesGraph, String origin, String destination) {
```

Comment:
Why did you decide that this class needed to be a service and not a utility class?

Reply:
Moving this function into a service was a recommendation from my last week mentor, i didn't think about it too much. Should I change it?

Reply:
A spring-managed class usually interacts with other spring components like @Service or @Repository, that's why they need to be annotated, so that spring scans them up during the app's start up. If your class works well without spring sugar, then just make it a simple java class.
Nit: when you're only defining static methods, declare a private constructor, so your class can't be instantiated.

Reply:
apenas voy viendo esto, se me hace que fue confusión, este fue el comment:
`Esto esta bien, pero considera mover estos métodos de esta clase de utilidad a default methods en la interface, de esa manera tienes una relación semántica/jerárquica entre los métodos de utilidad y la clase que los requiere, en cambio si están en una clase aparte no es explícito para alguien usando tu código que ambas clases están asociadas, o que ya existen métodos que hacen algo que tal vez tu necesites pero en otro lugar.`
No que lo movieras a un Servicio sino a la interface del servicio como un método default o estático.

Reply:
Ah no, la recomendación me la hizo otro de mis mentores, sorry por no especificar. Fue para separar las responsabilidades de las clases